pcdsdevices/targets.py

```diff
@@ -260,3 +264,122 @@ def move(self, nxspaces, nyspaces, wait=False):
+            return
+
+    def get_presets(self):
+        """
+        Get the saved presets if any.
+
+        Returns
+        -------
+        pos : tuple
+            Three coordinate positions - bottom_left, top_left, top_right
+        """
+        try:
+            pos = [self.x.presets.positions.x_bottom_left.pos,
+                   self.y.presets.positions.y_bottom_left.pos,
+                   self.x.presets.positions.x_top_left.pos,
+                   self.y.presets.positions.y_top_left.pos,
+                   self.x.presets.positions.x_top_right.pos,
+                   self.y.presets.positions.y_top_right.pos]
+            bottom_left = (pos[0], pos[1])
+            top_left = (pos[2], pos[3])
+            top_right = (pos[4], pos[5])
+            return bottom_left, top_left, top_right
+        except Exception:
+            logger.warning('Could not get presets, try to set_presets.')
+
+    def map_points(self):
+        """
+        Map all the sample positions in 2-d coordinates.
+
+        Returns
+        -------
+        coord : list
+            List of all coordinate points for samples on the grid.
+        """
+        bottom_left, top_left, top_right = self.get_presets()
+        if all([bottom_left, top_left, top_right]) is None:
+            logger.warning('Could not get presets, make sure you set presets')
+            return
+        # distance from bottom_left to top_left
+        height = np.sqrt(np.power((top_left[0] - bottom_left[0]), 2)
+                         + np.power((top_left[1] - bottom_left[1]), 2))
+        # distance from top_left to top_right
+        width = np.sqrt(np.power((top_right[0] - top_left[0]), 2)
+                        + np.power((top_right[1] - top_left[1]), 2))
+
+        # the very first coordinate is the most top-left sample on the grid
+        current_x = top_left[0]
+        current_y = top_left[1]
+        # we need to get all the places where the samples will be on both axes
+        x_grid_points = [current_x]
+        y_grid_points = [current_y]
+
+        # starting at the top left corner of the grid
+        # find all the samples locations on the x axes
+        while current_x < width - self.x_spacing:
+            current_x = current_x + self.x_spacing
+            x_grid_points.append(current_x)
+
+        # starting at the top left corner of the grid
+        # find all the samples locations on the y axes
+        while current_y < height + self.y_spacing:
+            current_y = current_y + self.y_spacing
+            y_grid_points.append(current_y)
+
+        # The meshgrid function returns
+        # two 2-dimensional arrays
+        xx, yy = np.meshgrid(x_grid_points, y_grid_points,
+                             sparse=False, indexing='xy')
+
+        plt.plot(xx, yy, marker='.', color='k', linestyle='none')
+        plt.show()
+
+        # flat out the arrays of points
+        flat_xx = list(chain.from_iterable(xx))
```

Comment:
This flattening process is not placing the points in the list the way we want..... (snake-style)

So I think I should not be flattening the 2d arrays here at all, and return a 2d arrays for x points, and a 2d array for y points? and deal with the rest in the scan function. (`nabs`)

Reply:
Mind pasting the plot here? Haven't gotten around to running this yet

Reply:
This is the plot that I get from here:
```
    xx, yy = np.meshgrid(x_grid_points, y_grid_points,
                             sparse=False, indexing='xy')

        plt.plot(xx, yy, marker='.', color='k', linestyle='none')
        plt.show()
```
My presets are these:
`bottom_left`: `(-100, 1316)`
`top_left`: `(-100, -100)`
`top_right`: `(2036, -100)`

This plot is only showing the grid points:
![point_plot](https://user-images.githubusercontent.com/62306310/102279049-a7df7200-3edf-11eb-86d0-f1e96ae0483f.PNG)

And this one is with this guy:
```
plan = list_scan([det], motor1, xx, motor2, yy)
plot_raster_path(plan, 'motor1', 'motor2', probe_size=.001)
```
![scan_points](https://user-images.githubusercontent.com/62306310/102279216-e07f4b80-3edf-11eb-8c5a-43be3cea1594.PNG)

not looking too good

Reply:
With the addition of `get_snake_grid_list` in ba1e39f I get something like in this picture below, this function is not too neat, i'm thinking that python must have something better for such thigs...
![snake_like](https://user-images.githubusercontent.com/62306310/102285496-40c7ba80-3eeb-11eb-82ba-655a75b3fb9f.PNG)

Reply:
There may be something that could serve as a good example in bluesky itself for cleaning up the snaking code. 

Perhaps you could use bluesky's snaking code calculations, then fix up the resulting points based on the transformation matrix determined by the user-selected points? Easier said than done, of course...